### PR TITLE
Added vim-swoop to programming layer. 

### PR DESCRIPTION
**Pull request by yzhuang**
Same as helm-swoop in spacemacs, invoked with <space>-s

This change makes space-vim behavior closer to spacemacs, by adding functionality similar to helm-swoop.

### CODE REVIEW

File: layers/+vim/programming/config.vim [hunk 1]
```diff
@@ -157,3 +157,14 @@ augroup END
...
+
+" vim-swoop {
+    " Do not let vim-swoop create keybindings
+    let g:swoopUseDefaultKeyMap = 0
+    " Visual selection mode mapping: swoop current selection.
+    vmap <Leader>s :call SwoopMultiSelection()<CR>
+    " Normal mode mapping: <Leader>s to start swoop buffer.
+    nnoremap <Leader>s :call SwoopMulti()<CR>
+    " normal mode mapping: select current word, then swoop.
+    nnoremap <Leader>ss viw:call SwoopMultiSelection()<CR>
```
Comment (liuchengxu) [inline layers/+vim/programming/config.vim]: `<Leader>s` and `<Leader>ss` could result in a mapping delay. When you type `<Leader>s`, vim will judge whether what you want to input is `<Leader>s` or actually `<Leader>ss`. See `:h timeoutlen`.

**Comment by liuchengxu** [comment]
Thanks, [vim-swoop](pelodelfuego/vim-swoop) is nice, but I think we should give it more intuitive key mappings. In fact, I have never used helm-swoop in emacs, so I'm not very familiar with it :).

**Comment by yzhuang** [comment]
Updated branch to remove the timeout delay from binding both \<leader\>s and \<leader\>ss in normal mode.  Now the keybinding is the same as Spacemacs.

**Comment by liuchengxu** [comment]
@yzhuang  How about this?

```vim
nnoremap <Leader>sm :call SwoopMulti()<CR>
```

We don't have to be exactly same to spacemacs actually :).

**Comment by yzhuang** [comment]
Agree.  \<Leader\>sS is difficult to type. 
Changed to \<Leader\>sm as suggested.

**Comment by liuchengxu** [comment]
@yzhuang Shall we squash the commits in this PR? I think it's ready to be merged.

**Comment by yzhuang** [comment]
Do you mind clicking the "squash and merge pull requests" button? 

Or you would like me to squash the commits into one on my local branch?

**Comment by liuchengxu** [comment]
Thanks! I didn't notice that option  🙉 .

**Pull request merged**